### PR TITLE
Feature/time event#10

### PR DESCRIPTION
**Pull request by iowusu**
Added time base waits as define in ASL.

### CODE REVIEW

File: state-conductor-example/src/test/ml-modules/root/test/suites/StateConductorSuite/test-data/flows/test-time-wait.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      "Type": "Task",
+      "Comment": "makes the envelope's triples",
+      "Resource": "/state-conductor/actions/custom/make-headers",
+      "Next": "success"
+    },
+    "success": {
+      "Type": "Succeed"
+    }
+  }
+}
```
Comment (aclavio) [inline state-conductor-example/src/test/ml-modules/root/test/suites/StateConductorSuite/test-data/flows/test-time-wait.json]: It probably doesn't matter as this is a test document, but should this use the `.asl.json` file extension?

File: state-conductor-example/src/test/ml-modules/root/test/suites/StateConductorSuite/timeWait-test.sjs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    timestamp: "2020-03-17T13:15:00Z",
+    nextTaskTime: "2020-03-17T13:15:00-04:00"
+  }
+});
+
+//"found waiting job")
+assertion = sc.resumeWaitingJobByJobDoc(jobDoc, "waitTask", false);
+assertions.push(test.assertEqual("working", assertion.flowStatus, "working flowStatus"));
+assertions.push(test.assertEqual("needs-envelope",assertion.provenance[3].to, "went to next state"));
+
```
Comment (aclavio) [inline state-conductor-example/src/test/ml-modules/root/test/suites/StateConductorSuite/timeWait-test.sjs]: add an assert here that also checks `assertion.flowState === "needs-envelope"`

Comment (aclavio) [inline state-conductor-example/src/test/ml-modules/root/test/suites/StateConductorSuite/timeWait-test.sjs]: add an assert here that verifies that the "currentlyWaiting" block has been removed

File: state-conductor-example/src/test/ml-modules/root/test/suites/StateConductorSuite/timeWait-test.sjs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  createdDate: "2020-03-17T21:41:19.022Z",
+  context: {},
+  provenance: [],
+  errors: {}
+});
+
+//"test excute job")
+assertion = sc.executeStateByJobDoc(jobDoc, false);
+assertions.push(test.assertEqual("waiting", assertion.flowStatus, "waiting flowStatus"));
+
```
Comment (aclavio) [inline state-conductor-example/src/test/ml-modules/root/test/suites/StateConductorSuite/timeWait-test.sjs]: add an assert here that verifies a "currentlyWaiting" block has been added and contains the proper contents

File: state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs [hunk 4]
```diff
@@ -318,6 +319,8 @@ function startProcessingFlowByJobDoc(jobDoc, save = true) {
  * @param {*} uri - the job document's uri
  */
 function resumeWaitingJob(uri, resumeBy = 'unknonw', save = true) {
+//ike testing trace
+  xdmp.trace(TRACE_EVENT, `check if resumeWaiting is working "${uri}"`);
```
Comment (aclavio) [inline state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs]: if this is just for testing, lets remove it

File: state-conductor-modules/src/main/ml-modules/root/state-conductor/tasks/waitTask.sjs [hunk 5]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+declareUpdate();
+//this take runs every minutes and excutes docs that have elapsed scheduled wait time
```
Comment (aclavio) [inline state-conductor-modules/src/main/ml-modules/root/state-conductor/tasks/waitTask.sjs]: typo: take => task

File: state-conductor-example/src/test/ml-modules/root/test/suites/StateConductorSuite/setup.sjs [hunk 6]
```diff
@@ -9,9 +9,10 @@ const test = require('/test/test-helper.xqy');
...
 test.loadTestFile('test-doc3.json', xdmp.database(), '/data/test-doc3.json');
+test.loadTestFile('test-doc4.json', xdmp.database(), '/data/test-doc4.json');
 test.loadTestFile('lorem.txt', xdmp.database(), '/data/lorem.txt');
 
 xdmp.documentAddCollections('/data/test-doc2.json', [sc.FLOW_ITEM_COLLECTION, 'test']);
 xdmp.documentAddCollections('/data/test-doc3.json', [sc.FLOW_ITEM_COLLECTION, 'enrollee']);
-
-test.log('StateConductorSuite Test Setup COMPLETE....');
\ No newline at end of file
+xdmp.documentAddCollections("/data/test-doc4.json", [sc.FLOW_ITEM_COLLECTION,"waitStateTest"]);
```
Comment (freshie) [inline state-conductor-example/src/test/ml-modules/root/test/suites/StateConductorSuite/setup.sjs]: (also for line 14 and 15

use the test.loadTestFile() argument for setting the collection 
example:
test.loadTestFile(fileName, databaseID, URI, Permissions, Collections);

test.loadTestFile('flows/test-time-wait.asl.json', xdmp.database(), sc.FLOW_DIRECTORY + 'test-time-wait.asl.json', xdmp.defaultPermissions(), [sc.FLOW_COLLECTION, "waitStateTest"]);

**Pull request merged**